lib/protocol/Result.js

```diff
@@ -76,41 +76,36 @@ Result.prototype.getLobColumnNames = function getLobColumnNames() {
 };
 
 Result.prototype.handle = function handle(err, reply, cb) {
-  if (err) {
-    return cb(err);
-  }
+  var reply = reply || {};
```

Comment:
With this change error handling and reply handling become mixed which adds some complexity. In the `switch-case` block there might be an error or there might be not + there might be a real reply or there might be an empty object and the code of every `case` should be 'aware' of that. How about something like:

```js
  if (err && reply) {
    if ([FunctionCode.INSERT, FunctionCode.UPDATE, FunctionCode.DELETE].indexOf(reply.functionCode) > -1) {
      return cb(err, reply.rowsAffected);
    }
  }
  if (err) {
    return cb(err);
  }
```

Then the rest of the code will stay the same.

It might even be possible (but I am not complete sure about it) to simplify the first check to the following:

```js
  if (err && reply && reply.rowsAffected) {
    return cb(err, reply.rowsAffected);
  }
```
But this is only possible if the rest of the operations (function codes) do not have the `rowsAffected` property and `rowsAffected` is an array. Better use the longer and safer  check.